### ISSUE TITLE
Numerous tests don't need to run entire bin/uglifyjs binary

### ISSUE THREAD

**Issue by fabiosantoscode** [terser-js/terser#114]
**Testing improvement**

Numerous tests rely on starting bin/uglifyjs binary just to exercise lib/parse.js:

```
    ✓ Should throw syntax error (5--) (199ms)
    ✓ Should throw syntax error (Math.random() /= 2) (206ms)
    ✓ Should throw syntax error (++this) (202ms)
    ✓ Should throw syntax error (++null) (198ms)
    ✓ Should throw syntax error (a.=) (197ms)
    ✓ Should throw syntax error (%.a) (195ms)
    ✓ Should throw syntax error (a./();) (197ms)
    ✓ Should throw syntax error ({%: 1}) (197ms)
    ✓ Should throw syntax error (const a) (200ms)
    ✓ Should throw syntax error (delete x) (204ms)
    ✓ Should throw syntax error (function g(arguments)) (201ms)
    ✓ Should throw syntax error (function eval()) (197ms)
    ✓ Should throw syntax error (iife arguments()) (201ms)
    ✓ Should throw syntax error (catch(eval)) (200ms)
    ✓ Should throw syntax error (var eval) (199ms)
    ✓ Should throw syntax error (else) (197ms)
    ✓ Should throw syntax error (return) (196ms)
    ✓ Should throw syntax error (block-level export) (234ms)
    ✓ Should throw syntax error (block-level import) (199ms)
    ✓ Should throw syntax error (anonymous class) (199ms)
    ✓ Should throw syntax error (anonymous function) (195ms)
    ✓ Should throw syntax error (spread in sequence) (197ms)
    ✓ Should throw syntax error (for-in init) (218ms)
    ✓ Should throw syntax error (for-in var) (197ms)
```

These can be replaced with tests that run in-process instead of spawning terser as a process, making them smaller and faster too

**Comment by kzc** [comment]
Those are intentionally testing the informative CLI errors with the caret (^) character which are not present when using the Terser API directly.

```
$ echo 'var x = 5--;' | terser 
Parse error at 0:1,9
var x = 5--;
         ^
ERROR: Invalid use of -- operator
```

How could the CLI be tested in-process?

**Comment by fabiosantoscode** [comment]
I think it's cool to test the caret is in the right place, but once you've tested one, you've tested them all, I think. 

You wouldn't be testing the CLI in-process, but instead testing the parse functionality

**Comment by kzc** [comment]
Fair point for the syntax errors.

The ones testing the CLI directly or special cases for the caret errors should be kept, such as:

https://github.com/fabiosantoscode/terser/blob/master/test/mocha/cli.js#L233-L271

I think the `test/compress/` syntax should be extended to handle syntax errors:

```
test_syntax_error: {
    input: `
        // notice the code is within a template string
        // as opposed to a block so that the test can
        // survive a parse error
        var x = 5--;
    `
    expect_error: {
        type: "SyntaxError",
        message: "Invalid use of -- operator",
        line: 6,
        col: 17,
    }
}
```
Mind you, this is not a task for a beginner. You'd need in-depth knowledge of the Terser AST to extend `test/run-tests.js` to handle that new syntax.

**Comment by kzc** [comment]
Invalid code within a template string has the obvious problem of not being able to handle backtick characters and template expressions without being escaped.

**Comment by kzc** [comment]
`expect_error` implemented in #115
